src/extensions/default/analyzers/Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes/HttpContextRefactor/CSharpHttpContextRefactorCodeFixProvider.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System.Composition;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.DotNet.UpgradeAssistant.Extensions.Default.CodeFixes
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = nameof(CSharpHttpContextRefactorCodeFixProvider))]
+    [Shared]
+    public class CSharpHttpContextRefactorCodeFixProvider : HttpContextRefactorCodeFixProvider<InvocationExpressionSyntax, ArgumentSyntax>
+    {
+        private protected override bool IsEnclosedMethodOperation(IOperation operation)
+            => operation is IMethodBodyOperation;
```

Comment:
Can we just check whether the operation is `CS.IMethodBodyOperation` or `VB.IBlockOperation` in the base class so that we only have one code fix provider? I don't mind have a base class with two language-specific children if we need to, but these differences are so small it seems like it may be better to just have separate code paths in these two places in the base class.

Reply:
Done. Using extension methods